### PR TITLE
return HTTP_CONFLICT 409 on Doctrine OptimisticLockException

### PR DESCRIPTION
**Pull request by bendavies**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes?
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | =
| License       | MIT
| Doc PR        | 

Seems like a nice sane default for Doctrine. thoughts?
I needed this in my API anyway.

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -259,11 +261,13 @@ private function addExceptionToStatusSection(ArrayNodeDefinition $rootNode)
...
-                        ExceptionInterface::class => Response::HTTP_BAD_REQUEST,
-                        InvalidArgumentException::class => Response::HTTP_BAD_REQUEST,
-                        FilterValidationException::class => Response::HTTP_BAD_REQUEST,
-                    ])
+                    ->defaultValue(
+                        array_merge([
+                            ExceptionInterface::class => Response::HTTP_BAD_REQUEST,
+                            InvalidArgumentException::class => Response::HTTP_BAD_REQUEST,
+                            FilterValidationException::class => Response::HTTP_BAD_REQUEST,
+                        ], class_exists(Version::class) ? [OptimisticLockException::class => Response::HTTP_CONFLICT] : [])
```
Comment (bendavies) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php]: using `Version::class` as that is the convention.

**Comment by antograssiot** [comment]
Seems a bit app specific to me ...

File: src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (dunglas) [reply src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php]: Looks useless, PHP will allow `OptimisticLockException::class` even if the class doesn't actually exist. It doesn't hurt to have this extra key.

Comment (bendavies) [reply src/Bridge/Symfony/Bundle/DependencyInjection/Configuration.php]: done

**Comment by dunglas** [comment]
Thanks @bendavies

**Comment by bendavies** [comment]
thanks @dunglas

**Pull request merged**